core/modules/parser/QSMySqlListener.cc

```diff
@@ -2462,6 +2474,31 @@ void QSMySqlListener::exitRoot(QSMySqlParser::RootContext* ctx) {
 }
 
 
+std::string QSMySqlListener::getStringTree() const {
+    auto ldh = _listenerDebugHelper.lock();
+    if (ldh != nullptr) {
```

Comment:
I thought we agreed that shared pointer implicit conversion to bool was not good. Was that only in the context of boost unit tests?

"very Pythonic" does not help your case (with me) 😆

Reply:
No, I didn't imply we should revisit the previous conversation on the implicit conversion to bool :-) I adopted your suggestion to avoid this. My only point was on some (a bit) extra safety in reverse order of terms in the logical comparison. There are other reasons to do this which I mentioned in some ither comment.